### ISSUE TITLE
Regression after a BIND/dnsutils update "systemd-resolved[1495]: DNSSEC validation failed"

### ISSUE THREAD

**Issue by GaalexxC** [systemd/systemd#30041]
Seems to be a regression after a BIND/dnsutils update approx 2 weeks ago [BIND 1:9.16.1-0ubuntu2.16/9.18.18-0ubuntu0.22.04.1-Ubuntu (Extended Support Version)], effecting Ubuntu versions 20x/22x. I have had the issue since the update. The longer it is allowed to persist the worse it gets to the point DNS errors and failures show in all logs, Nginx, domain, mail server, and even apt when running an update.

**In reference original Issue**
https://github.com/systemd/systemd/issues/6490#

`systemd 249 (249.11-0ubuntu3.11)`

> Nov 14 09:21:15 thundercloud systemd-resolved[1495]: DNSSEC validation failed for question . IN SOA: incompatible-server
> Nov 14 09:21:15 thundercloud systemd-resolved[1495]: DNSSEC validation failed for question arpa IN DS: incompatible-server
> Nov 14 09:21:15 thundercloud systemd-resolved[1495]: DNSSEC validation failed for question arpa IN SOA: incompatible-server
> Nov 14 09:21:15 thundercloud systemd-resolved[1495]: DNSSEC validation failed for question in-addr.arpa IN DS: incompatible-server
> Nov 14 09:21:15 thundercloud systemd-resolved[1495]: DNSSEC validation failed for question in-addr.arpa IN SOA: incompatible-server
> Nov 14 09:21:15 thundercloud systemd-resolved[1495]: DNSSEC validation failed for question 94.in-addr.arpa IN DS: incompatible-server
> Nov 14 09:21:15 thundercloud systemd-resolved[1495]: DNSSEC validation failed for question 156.94.in-addr.arpa IN DS: incompatible-server
> Nov 14 09:21:15 thundercloud systemd-resolved[1495]: DNSSEC validation failed for question 156.94.in-addr.arpa IN SOA: incompatible-server
> Nov 14 09:21:15 thundercloud systemd-resolved[1495]: DNSSEC validation failed for question 69.156.94.in-addr.arpa IN DS: incompatible-server
> Nov 14 09:21:15 thundercloud systemd-resolved[1495]: DNSSEC validation failed for question 69.156.94.in-addr.arpa IN SOA: incompatible-server
> Nov 14 09:21:15 thundercloud systemd-resolved[1495]: DNSSEC validation failed for question 31.69.156.94.in-addr.arpa IN DS: incompatible-server
> Nov 14 09:21:15 thundercloud systemd-resolved[1495]: DNSSEC validation failed for question 31.69.156.94.in-addr.arpa IN SOA: incompatible-server
> Nov 14 09:21:15 thundercloud systemd-resolved[1495]: DNSSEC validation failed for question 31.69.156.94.in-addr.arpa IN PTR: incompatible-server
> Nov 14 09:21:15 thundercloud systemd-resolved[1495]: DNSSEC validation failed for question . IN SOA: incompatible-server
> Nov 14 09:21:15 thundercloud systemd-resolved[1495]: DNSSEC validation failed for question arpa IN DS: incompatible-server
> Nov 14 09:21:15 thundercloud systemd-resolved[1495]: DNSSEC validation failed for question arpa IN SOA: incompatible-server
> Nov 14 09:21:15 thundercloud postfix/smtpd[78707]: connect from unknown[94.156.69.31]
> Nov 14 09:21:15 thundercloud systemd-resolved[1495]: DNSSEC validation failed for question in-addr.arpa IN DS: incompatible-server
> Nov 14 09:21:15 thundercloud systemd-resolved[1495]: DNSSEC validation failed for question in-addr.arpa IN SOA: incompatible-server
> Nov 14 09:21:15 thundercloud systemd-resolved[1495]: DNSSEC validation failed for question 94.in-addr.arpa IN DS: incompatible-server
> Nov 14 09:21:15 thundercloud systemd-resolved[1495]: DNSSEC validation failed for question 31.69.156.94.in-addr.arpa IN PTR: incompatible-server

`service systemd-resolved restart`  resets it but lasts less than an hour, so I run an hourly cron for now. I run forums and media applications, all uploads fail to S3 buckets with same errors in logs if not reset. apt update fails as well and cannot connect to repos if not reset in timely manner.

_Originally posted by @GaalexxC in https://github.com/systemd/systemd/issues/6490#issuecomment-1810321223_

**Comment by TriMoon** [comment]
Are you sure your BIND-DNS server is working correctly?
If you use BIND as a local DNS server, you should point your system's DNS to use it, and have BIND forward requests to other servers.

Otherwise you risk that your system gets conflicting answers.

That said:
I personally have no idea what changed in BIND in the last 2 weeks which makes it "incompatible"...

**Comment by pemensik** [comment]
No, recent *bind9* cannot be configured to not provide *DNSSEC records* since BIND 9.12. I doubt bind is responsible, but I guess more details is required to fix the issue. Can you record dns traffic using wireshark or tcpdump on ``port domain``? What does command ``delv @${NAMED_IP}``, where you specify IP of your forwarder, print?

Please share at least ``resolvectl`` output. If you want it fixed, developer should not have to dig for details in nested information. At least basic information is required here in the issue.
